File: tests/cpp/greedy_algorithms/test_minimum_waiting_time.cpp
#include "../../../problems/greedy_algorithms/minimum_waiting_time/cpp/minimum_waiting_time.h"
#include "../jsontestbase.h"

#include <vector>

class MinimumWaitingTimeTest : public JsonTestBase {
public:
  MinimumWaitingTimeTest() {
    json_file_path = std::string(TEST_CASES_DIR) +
                     "/greedy_algorithms/minimum_waiting_time.json";
  }
};

TEST_F(MinimumWaitingTimeTest, TestCases) {
  ASSERT_FALSE(test_cases.empty()) << "Test cases are empty or not loaded";

  for (const auto &test : test_cases) {
    auto queries = test["queries"].get<std::vector<int>>();
    auto expected = test["expected"].get<int>();

    auto result = minimumWaitingTime(queries);

    EXPECT_EQ(result, expected) << "Failed for test case: " << test.dump(2);
  }
}

File: tests/python/greedy_algorithms/minimum_waiting_time/test_minimum_waiting_time.py
from problems.greedy_algorithms.minimum_waiting_time.python.minimum_waiting_time import \
    minimumWaitingTime
from tests.python.utility import load_test_cases


def test_minimum_waiting_time() -> None:
    test_cases = load_test_cases("greedy_algorithms/minimum_waiting_time.json")

    for idx, test in enumerate(test_cases):
        queries: list[int] = test["queries"]
        expected: int = test["expected"]

        result: int = minimumWaitingTime(queries)

        assert (
            result == expected
        ), f"Test case {idx} failed: expected {expected}, got {result}"
